src/fs.h

```diff
@@ -21,6 +21,15 @@ namespace fs = boost::filesystem;
 namespace fsbridge {
     FILE *fopen(const fs::path& p, const char *mode);
 
+    /**
+     * Helper function for joining paths
+     *
+     * @param[in] base  Absolute path
+     * @param[in] path  Path to combine with base
+     * @returns path if it is an absolute path, otherwise base joined with path
```

Comment:
In commit "Replace fs::absolute calls with AbsPathJoin calls" (215c78465059175505218566f2ec38d52d3be827)

Maybe expand this line a little to say "@returns path unchanged if it is an absolute path, otherwise returns base joined with path. Returns base unchanged if path is empty."

If desired, could also add preconditions and postconditions (base path must be absolute, returned path will always be absolute).

The information about paths being unchanged could be useful to callers and would be hard to discover without digging into the implementation and library documentation. It also explains how this join is different than other possible joins.